File: Sorting/insertionSort.ts
const selectionSort = (arr) => {
  for (let i = 1; i < arr.length; i++) {
    while (i> 0 && arr[i-1] > arr[i]) {
      [arr[i], arr[i-1]] = [arr[i-1],arr[i]]
      i--
    }
  }
  return arr;
};
const a = [12,12,1,1,1,1,39,27,24,26,0,26,35,24,46];
console.log(selectionSort(a));

File: Sorting/insertionSort.java
class Main {
    static int[] insertionSort(int arr[]) {
        for (int i = 1; i < arr.length; i++) {
            int j;
            j = i - 1;
            while (j >= 0 && arr[j] > arr[j + 1]) {
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
                j--;
            }
        }
        return arr;
    }

    public static void main(String[] args) {
        int a[] = {92, 39, 27, 84, 26, 0, 26, 95, 24, 56};
        int new_array[] = insertionSort(a);
        for (int num : new_array) {
            System.out.print(num + " ");
        }
    }
}
